### PR TITLE
 let checkpoint data stored as flat file by default

### PR DESCRIPTION
**Pull request by raindust**

### CODE REVIEW

File: dpos/state/checkpoint.go [hunk 1]
```diff
@@ -472,6 +493,111 @@ func NewRewardData() *RewardData {
...
+	}
+
+	if err = c.NextReward.Deserialize(r); err != nil {
+		return
+	}
+
+	return c.StateKeyFrame.Deserialize(r)
+}
+
+func (c *CheckPoint) persistBytesArray(value io.Writer,
```
Comment (yz89) [inline dpos/state/checkpoint.go]: Is it better to call it writeBytesArray ?

File: dpos/state/checkpoint.go [hunk 2]
```diff
@@ -472,6 +493,111 @@ func NewRewardData() *RewardData {
...
+
+	for _, b := range bytesArray {
+		if err = common.WriteVarBytes(value, b); err != nil {
+			return
+		}
+	}
+	return
+}
+
+func (c *CheckPoint) getBytesArray(r io.Reader) ([][]byte, error) {
```
Comment (yz89) [inline dpos/state/checkpoint.go]: so it is called readBytesArray

File: dpos/state/checkpoint.go (hunk 1 above)
Comment (raindust) [reply dpos/state/checkpoint.go]: Ok, fixed it

File: dpos/state/checkpoint.go (hunk 2 above)
Comment (raindust) [reply dpos/state/checkpoint.go]: Fixed it

**Pull request merged**